tempodb/tempodb.go

```diff
@@ -544,3 +544,33 @@ func (rw *readerWriter) compactedBlocklist(tenantID string) []*encoding.Compacte
 
 	return copiedBlocklist
 }
+
+// updateBlocklist Add and remove regular or compacted blocks from the in-memory blocklist.
+// Changes are temporary and will be overwritten on the next poll.
+func (rw *readerWriter) updateBlocklist(tenantID string, add []*encoding.BlockMeta, remove []*encoding.BlockMeta, compactedAdd []*encoding.CompactedBlockMeta) {
+	if tenantID == "" {
+		return
+	}
+
+	rw.blockListsMtx.Lock()
+	defer rw.blockListsMtx.Unlock()
+
+	// ******** Regular blocks ********
+	toRemove := make(map[*encoding.BlockMeta]bool)
+	for _, m := range remove {
+		toRemove[m] = true
+	}
+
+	blocklist := rw.blockLists[tenantID]
+	newblocklist := make([]*encoding.BlockMeta, 0, len(blocklist)-len(remove)+len(add))
```

Comment:
concerned about this: `len(blocklist)-len(remove)+len(add)`

when writing tests that target this method try some wacky things like a len that comes back negative.  i know that shouldn't be possible but i think we should guard against it if it panics

Reply:
Thanks, did work out an issue where it did not handle correctly attempting to remove blocks that didn't exist. Fixed and covered with tests.